onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -275,6 +276,12 @@ InlinedVector<std::unique_ptr<GraphTransformer>> GenerateTransformers(
         transformers.emplace_back(std::make_unique<GeluApproximation>(cpu_cuda_rocm_eps));
       }
 
+#ifdef ENABLE_TRAINING
+      const std::string enable_memory_alleviation =
```

Comment:
This could be a little bit tricky because knowing fw graph, we cannot know exactly whether one activation preserved or not in backward, so we cannot get candidates.  In ORTTrainer recompute implementation, we did against fw graph, assuming if the pre-defined subgraph pattern happens, we will recompute them, while this rule sometimes is too general imagine if Gelu is one of the patterns, then all Gelu will be recomputed.

Reply:
Oh, I did not realize what benefit we can get doing during gradient graph building. Do you think there will be more changes to make the implementation better?

Reply:
I was thinking the benefit is that we can build the recompute logic at the time of gradient graph building, instead of building the gradient graph and then creating replacing the backward nodes to add recompute nodes (saving initialization time; not that it matters a lot for `ORTModule` scenario).
The other benefit is that we don't need to then rely on post `InferenceSession` graph transformations, and the code can be restricted to the orttraining subdir only.

I realize that these points are probably not very important, so I guess this is not required right now. Just curious if this was something you considered.